src/main/java/org/apache/commons/io/filefilter/AbstractFileFilter.java

```diff
@@ -37,7 +37,13 @@
      * @return true if this file matches the test
      */
     @Override
+    @SuppressWarnings("nullness:argument.type.incompatible")
     public boolean accept(final File file) {
+        // File.getParentFile() can return null, and null is not documented
+        // as a permitted file argument for FilenameFilter.accept(File,String).
+
+        // However following code does not lead to any Exception at run time
```

Comment:
I have already reported this bug. [Bug report](https://issues.apache.org/jira/browse/IO-540?jql=project%20%3D%20IO%20AND%20status%20in%20(Open%2C%20%22In%20Progress%22%2C%20Reopened)%20ORDER%20BY%20key%20DESC) But I was later confused about their documentation. I request you to suggest me changes/improvements in my bug report, I will try to produce test-case.